internal/api/lock.go

```diff
@@ -150,3 +150,38 @@ func lockInstallationBackup(c *Context, backupID string) (*model.InstallationBac
 		})
 	}
 }
+
+// lockInstallationDBMigrationOperation synchronizes access to the given db migration operation across
+// potentially multiple provisioning servers.
+func lockInstallationDBMigrationOperation(c *Context, operationID string) (*model.InstallationDBMigrationOperation, int, func()) {
+	dbMigrationOperation, err := c.Store.GetInstallationDBMigrationOperation(operationID)
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to query db migration operation")
+		return nil, http.StatusInternalServerError, nil
+	}
+	if dbMigrationOperation == nil {
+		return nil, http.StatusNotFound, nil
+	}
+
+	locked, err := c.Store.LockInstallationDBMigrationOperation(operationID, c.RequestID)
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to lock db migration operation")
+		return nil, http.StatusInternalServerError, nil
+	} else if !locked {
+		c.Logger.Error("failed to acquire lock for db migration operation")
+		return nil, http.StatusConflict, nil
+	}
+
+	unlockOnce := sync.Once{}
+
+	return dbMigrationOperation, 0, func() {
```

Comment:
Since this method uses HTTP status codes to signal its error status when it returns, would it make more sense to return an HTTP success code (OK or Accepted) here instead of 0? When reading the call sites of this function, it wasn't clear to me what the 0 return code that we were checking for meant, and I feel that the mix of HTTP status codes with a Unix/C convention (0 for success) is a bit inelegant.

Reply:
This is the pattern we use in all lock methods in this file, for example https://github.com/mattermost/mattermost-cloud/blob/master/internal/api/lock.go#L37.

I am 0/5 on changing it but I think consistency is more important so if we decide to go for it, we should change all similar lock methods.